designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/simple/SimpleStateStatus.scala

```diff
@@ -50,13 +51,30 @@ object SimpleStateStatus {
 
   }
 
-  val NotDeployed: StateStatus = AllowDeployStateStatus("NOT_DEPLOYED")
-  val DuringDeploy: StateStatus = DuringDeployStateStatus("DURING_DEPLOY")
-  val Running: StateStatus = RunningStateStatus("RUNNING")
-  val Finished: StateStatus = FinishedStateStatus("FINISHED")
-  val Restarting: StateStatus = NotEstablishedStateStatus("RESTARTING")
-  val DuringCancel: StateStatus = NotEstablishedStateStatus("DURING_CANCEL")
-  val Canceled: StateStatus = AllowDeployStateStatus("CANCELED")
+  val NotDeployed: StateStatus = new StateStatus {
```

Comment:
Let's create a companion object StateStatus with:

- apply(name: String)
- duringDeploy(name: String)
- running(name: String)
- finished(name: String)

these methods should allow reducing boilerplate.

Reply:
Added those methods to keep clarity and readibility.